packages/icon/src/vaadin-icon.js

```diff
@@ -226,6 +240,46 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(ConditionalResizeM
     }
   }
 
+  async __srcChanged(src) {
+    if (!src) {
+      return;
+    }
+
+    if (src.includes('#')) {
+      this.svg = svg`<use href="${src}"/>`;
+    } else {
+      try {
+        const svgData = await this.__fetchSVG(src);
+        if (!parser) {
+          parser = new DOMParser();
+        }
+        const parsedResponse = parser.parseFromString(svgData, 'text/html');
+
+        const svgElement = parsedResponse.querySelector('svg');
+        if (!svgElement) {
+          throw new Error(`SVG element not found on path: ${src}`);
```

Comment:
Untested code

Reply:
The tests still pass with this commented out.

Reply:
Added an assertion to verify the content of the `console.error` message